### PR TITLE
warn when running console instead of repl

### PR DESCRIPTION
**Pull request by lrytz**
image

-----

re https://github.com/lampepfl/dotty/pull/19174#issuecomment-1951344905

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -2140,6 +2140,10 @@ object Build {
         repl := (`scala3-compiler-bootstrapped` / repl).value,
       ).
       settings(
```
Comment (bishabosha) [inline project/Build.scala:2142]: maybe this should be in `nonBootstrappedSettings`?

`scala3-bootstrapped/console` will use the snapshot version
```suggestion
      nonBootstrappedSettings(
```

Comment (bishabosha) [reply project/Build.scala:2142]: imm not actually sure what `nonBootstrappedSettings` does, just saying that this warning only makes sense for console task in non-bootstrapped projects

**Review by bishabosha — APPROVED** [review]
I'd prefer a more "in your face" warning, but this works

**Comment by lrytz** [comment]
Happy to change it to whatever. There are plenty of emojis. Or make it an error?

**Comment by lrytz** [comment]
image

**Review by bishabosha — APPROVED** [review]

**Comment by SethTisue** [comment]
LGTM — I think the warning is better than an error, as it might still sometimes be handy to use `console` to see how the reference compiler handles something.

**Pull request merged**